### PR TITLE
bugfix/update financial transaction

### PR DESCRIPTION
**Pull request by sylvvester**
bugfix/update financial transaction  #214

### CODE REVIEW

**Review by Iwona007 — DISMISSED** [review]

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionServiceImpl.java [hunk 1]
```diff
@@ -89,10 +89,15 @@ public FinancialTransactionDTO updateFinancialTransaction(
...
-                            String.format("Financial transaction with id: %d not found", id));
-                });
+                .orElseThrow(() -> new AppRuntimeException(ErrorCode.FT001,
+                        String.format("Financial transaction with id: %d not found", id)));
+
+        if (uDTO.categoryId() != null) {
+            FinancialTransactionCategory financialTransactionCategory = financialTransactionCategoryRepository.findById(uDTO.categoryId())
+                    .orElseThrow(() -> new AppRuntimeException(ErrorCode.FTC001,
+                            String.format("Financial transaction category with id: %d does not exist", uDTO.categoryId())));
+            entity.setFinancialTransactionCategory(financialTransactionCategory);
```
Comment (mirekgab) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionServiceImpl.java:99]: Let's consider a situation where uDTO.type() != financialTransactionCategory.getType()?
Shouldn't an exception be thrown?

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionServiceImpl.java [hunk 2]
```diff
@@ -89,10 +89,15 @@ public FinancialTransactionDTO updateFinancialTransaction(
...
 
         FinancialTransaction entity = financialTransactionRepository.findById(id)
-                .orElseThrow(() -> {
-                    throw new AppRuntimeException(ErrorCode.FT001,
-                            String.format("Financial transaction with id: %d not found", id));
-                });
+                .orElseThrow(() -> new AppRuntimeException(ErrorCode.FT001,
+                        String.format("Financial transaction with id: %d not found", id)));
+
+        if (uDTO.categoryId() != null) {
```
Comment (dawciobiel) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionServiceImpl.java:95]: Please replace it with:

```java
Long categoryId = uDTO.categoryId();

        if (categoryId != null) {
            FinancialTransactionCategory financialTransactionCategory = financialTransactionCategoryRepository.findById(categoryId)
```

**Review by dawciobiel — CHANGES_REQUESTED** [review]
Changes required

File: src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionServiceImpl.java (hunk 2 above)
Comment (mirekgab) [inline src/main/java/pl/byczazagroda/trackexpensesappbackend/service/FinancialTransactionServiceImpl.java:95]: Consider a situation where entity.categoryId!=null and uDTO.categoryId==null. Will the categoryId be updated?

**Review by Iwona007 — APPROVED** [review]

**Review by dawciobiel — APPROVED** [review]
Looks good to me

**Pull request merged**